src/githubClient.ts

```diff
@@ -6,7 +8,35 @@ export class GitHubClient {
         this._token = token;
     }
 
-    public async createCheckRun(payload: any): Promise<void> {
+    public async createCheckRunWithFallback(checkRunThroughAppPayload: any, checkRunPayload: any): Promise<void> {
```

Comment:
Shouldn't "Fallback" be implicit / default expectation?

Reply:
Refactored as discussed offline.